de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/filesystem/VirtualFileConverter.java

```diff
@@ -0,0 +1,169 @@
+package de.fu_berlin.inf.dpp.intellij.filesystem;
+
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleUtil;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+
+import de.fu_berlin.inf.dpp.SarosPluginContext;
+import de.fu_berlin.inf.dpp.activities.SPath;
+import de.fu_berlin.inf.dpp.filesystem.IProject;
+import de.fu_berlin.inf.dpp.filesystem.IResource;
+
+import org.apache.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.picocontainer.annotations.Inject;
+
+/**
+ * Provides static methods to convert VirtualFiles to Saros resource objects
+ * or Saros resources objects to VirtualFiles.
+ */
+public class VirtualFileConverter {
+
+    private static final Logger log = Logger
+        .getLogger(VirtualFileConverter.class);
+
+    @Inject
+    private static Project project;
```

Comment:
What is the current status of the Project switching in the IntelliJ. IIRC it will break anything. Any plans to shutdown the session when switching projects or make the plugin work on multiple projects (aka IntelliJ application scope)

Reply:
Yes, changing the project still breaks everything. But I have not been looking at/working on this issue as I thought this was an acceptable restriction for a functioning alpha version in my mind.

I know that there was a discussion on the topic. The suggestion was to move the plugin back to an application context. I would agree that this is a sensible approach. This is probably a lot of work, but should be possible. We would have to figure out how to get the correct project object (and make it easily accessible somehow) and what to do if the shared module is open in multiple projects (as a module can be imported in any number of projects at once).

I am not sure how pressing this issue really is. But, since I don't have any big task lined up (well, I have #223, #261, and #285 but they hopefully should not take to long) after completing the rework of FileSystemChangeListener (which drags on forever; the logic has been done for a while but I am still struggling with how to split it up correctly and what to do in a few corner cases) to complete the filesystem rework, I could take a look at it.

As the basic functionality for Saros would be present (completely sharing a module and doing whatever you want inside it) and the missing stuff is not that important for an early version in my mind (follow mode, better UI, viewport annotations) or is somewhat blocked by waiting for the resource interface rework to remove IProject (sharing multiple modules and modules with multiple content roots), there does not seem to be a more pressing issue/feature to deal with in my opinion.

If you disagree, feel free to point me in the right direction on what needs to be fixed beforehand. Maybe I am just forgetting something. So many issues pop up when working on Saros/I that I sometimes just forget something I said I would do if I don't create an issue for it right away.

Reply:
Well I think this is not that urgent. If you present the user a message before (or even after) changing the project that something bad may happen should be ok for an Alpha Version.